man/sd_hwdb_get.xml

```diff
@@ -0,0 +1,173 @@
+    details.</para>
+
+    <para><function>sd_hwdb_seek()</function> selects entries matching the specified string
+    <parameter>modalias</parameter>. Subsequent queries with <function>sd_hwdb_enumerate()</function> will
+    access the key-value pairs for that string.</para>
+
+    <para><function>sd_hwdb_enumerate()</function> returns (in turn) all the key-value pairs defined for the
+    string used with <function>sd_hwdb_seek()</function>. Each pair is returned as
+    <constant>NUL</constant>-terminated strings in <parameter>key</parameter> and
+    <parameter>value</parameter>. The strings must not be modified by the caller and are valid as long as a
+    reference to <parameter>hwdb</parameter> is kept. When multiple patterns in the database match
+    <parameter>modalias</parameter>, the combination of all matching key-value pairs is used. See
+    <citerefentry><refentrytitle>hwdb</refentrytitle><manvolnum>7</manvolnum></citerefentry> for
+    details.</para>
+
+    <para>The <function>SD_HWDB_FOREACH_PROPERTY</function> macro combines
+    <function>sd_hwdb_seek()</function> and <function>sd_hwdb_enumerate()</function>. No error handling is
+    performed and interation simply stops on error. See the example below.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para>On success, <function>sd_hwdb_get()</function> and <function>sd_hwdb_seek()</function> return a
+    non-negative integer. On failure, they return a negative errno-style error code.</para>
+
+    <para><function>sd_hwdb_enumerate()</function> returns a positive integer if another key-value pair was found or zero if
+    all entries have already been enumerated. On failure, it returns a negative errno-style error code.
+    </para>
+
+    <refsect2>
+      <title>Errors</title>
+
+      <para>Returned errors may indicate the following problems:</para>
+
+      <variablelist>
+        <varlistentry>
+          <term><constant>-EINVAL</constant></term>
+
+          <listitem><para>A parameter is <constant>NULL</constant>.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-ENOENT</constant></term>
+
+          <listitem><para>An entry for the specified <parameter>modalias</parameter> was not found.
+          </para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-EAGAIN</constant></term>
+
+          <listitem><para><function>sd_hwdb_seek()</function> was not called before
+          <function>sd_hwdb_enumerate()</function>.</para></listitem>
+        </varlistentry>
+      </variablelist>
+    </refsect2>
+  </refsect1>
+
+  <xi:include href="libsystemd-pkgconfig.xml" />
+
+  <refsect1>
+    <title>Examples</title>
+
+    <example>
+      <title>Look up hwdb entries for a specific match string</title>
+
+      <programlisting>
+#include &lt;stdio.h&gt;
+#include &lt;sd-hwdb.h&gt;
+
+int lookup(char *match) {
```

Comment:
to make the example more real-lifey, maybe beef it up to do this:

```c
int lookup_usb_properties(uint16_t vid, uint16_t pid) {
        char match[15];
        /* Search fields matching by USB vendor/product IDs */
        snprintf(match, "usb:v%04Xp%04X", vid, pid);
        …    
}
```

(just an idea)